npm run build error

>npm run build
> AzureSearchIndexerDemo@0.0.1 build C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo
> browserify . > js/UI/bundle.min.js

path.js:7
    throw new TypeError('Path must be a string. Received ' + inspect(path));
    ^

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.dirname (path.js:695:5)
    at Deps.lookupPackage (C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\node_modules\module-deps\index.js:452:29)
    at Deps._transform (C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\node_modules\module-deps\index.js:93:10)
    at Deps.Transform._read (C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\node_modules\readable-stream\lib\_stream_transform.js:184:10)
    at Deps.Transform._write (C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\node_modules\readable-stream\lib\_stream_transform.js:172:12)
    at Labeled.Pipeline._write (C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\node_modules\stream-splicer\index.js:73:22)
    at doWrite (C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\node_modules\readable-stream\lib\_stream_writable.js:279:12)
    at writeOrBuffer (C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\node_modules\readable-stream\lib\_stream_writable.js:266:5)
    at Labeled.Writable.write (C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\node_modules\readable-stream\lib\_stream_writable.js:211:11)

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
npm ERR! node v6.9.2
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
npm ERR! AzureSearchIndexerDemo@0.0.1 build: `browserify . > js/UI/bundle.min.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the AzureSearchIndexerDemo@0.0.1 build script 'browserify . > js/UI/bundle.min.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the AzureSearchIndexerDemo package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     browserify . > js/UI/bundle.min.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs AzureSearchIndexerDemo
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls AzureSearchIndexerDemo
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Ed\Documents\GitHub\AzureSearchNodeJSIndexerDemo\npm-debug.log

--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,6 @@
 #node packages
 node_modules
+package-lock.json
 
 #bundles from build builds
 js/UI/bundle.*

--- a/package.json
+++ b/package.json
@@ -8,7 +8,7 @@
     "underscore": "^1.7.0"
   },
   "devDependencies": {
-    "browserify": "~6.2.0",
+    "browserify": "14.4.0",
     "envify": "~3.0.0",
     "react": "^0.12.0",
     "reactify": "^0.15",

